### PR TITLE
preprocessing of era5 sst data format changed for fv3net reservoir computing

### PR DESCRIPTION
**Pull request by paulaharder**
I changed era_process.py to
- have weekly mean quantities instead of daily
- take start and end dates as arguments to create a dataset for a specific time period
- not have a time dim issue that requires decode_times=False when loading with xarray
- have separate files for different tiles
- enable separate files for train/val/test
- include all given variables (sst,t2m,u10,v10,sst_tend) in one file
- shift t2m,u10,v10
- have a new field "sst_tendency", the sst tendency

### CODE REVIEW

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:75]: For visibility, hard code the list of variables that get shifted as `TIME_SHIFTED_VARIABLES` at the top of the module and refer to that variable here.

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:76]: Can just call `xr.DataArray.shift` directly here and remove the `shift_variable` function.

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:163]: Can define this netcdf name as a variable since it's used again later in the subsequent function call. For extra clarity I'd add a `weekly_mean` descriptor to the filenames.

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:79]: There's a number of temporary netcdfs that are written in this script before the final merged output is saved. For the intermediate netcdfs that are not needed, I would create them in a[ temporary directory](https://stackoverflow.com/questions/3223604/how-do-i-create-a-temporary-directory-in-python) so they are cleaned up at the end.

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:44]: I'd just omit this arg and have this information in the `out_path` arg instead.

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:62]: I missed this in the previous PR, but the download locations for the data saved in this function are also hard coded. I'd change this to save them in a temporary directory (see other comment about this).

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:90]: Can move these lines modifying attrs into the `merge_variables` function.

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:65]: Instead of calling the sequence of regridding and masking for each variable, can they all just be merged at the start? e.g. something like
```
for var in variables:
    check_time_steps_complete(...)
full_variables = merge_variables(path, variables)
# do time shift, add tendency by doing xarray operations on the full dataset, no need to write it in between steps
full_variables["sst_tendency"] = calculate_sst_tendency(...)
for var in TIME_SHIFT_VARIABLES:
    full_variables[var] = full_variables[var].shift(time=-1)
# masking, etc.
...
```

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:70]: With the exception of the netcdfs used as input to fregrid, you don't have to save netcfs in between operations. Instead, just pass the dataset to the next function and return the modified dataset.

**Review by AnnaKwa — CHANGES_REQUESTED** [review]
Thanks for the updates! High level comments:
- There are a few intermediate steps that save netcdfs when they could skip this and just pass a dataset to the next operation. I recommend removing these. For the cases where a temporary netcdfs does need to be written, i.e. the fregrid input data, I'd use a temporary directory that gets cleaned up after the script is done.
- If you move the merging of all the variables to the beginning of the operations, after the times are trimmed to the specified range, you can do the regridding, masking, etc. once and don't have to loop over variables

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:297]: The data loaders that are used in training take in a directory which is assumed to contain all the netcdfs for a single rank (tile). Could you change the output path to match this structure (one tile data per directory)?

File: projects/reservoir/fv3/era_process.py
Comment (paulaharder) [reply projects/reservoir/fv3/era_process.py:90]: Actually, I removed the attrs stuff

File: projects/reservoir/fv3/era_process.py
Comment (paulaharder) [reply projects/reservoir/fv3/era_process.py:70]: interpolated.nc of a leftover from debugging, I'll remove it

File: projects/reservoir/fv3/era_process.py
Comment (paulaharder) [reply projects/reservoir/fv3/era_process.py:65]: Maybe, but I was worried that fregrid is going to cause problems

File: projects/reservoir/fv3/era_process.py
Comment (paulaharder) [reply projects/reservoir/fv3/era_process.py]: I tried this, but it caused many issues with making directories or writing files, so I just created a /temp folder and removed it after

File: projects/reservoir/fv3/era_process.py
Comment (paulaharder) [reply projects/reservoir/fv3/era_process.py]: Okay, that#s done. It worked

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:105]: Move this line further up right after the temporary directory is created

File: projects/reservoir/fv3/era_process.py
Comment (AnnaKwa) [inline projects/reservoir/fv3/era_process.py:67]: Can be removed since now the reservoir training can save a mask array to deal with unwanted inputs. However since that would require additional changes to this PR, just leave this for now and add a TODO comment above it mentioning that it can be removed in a subsequent PR.

**Review by AnnaKwa — APPROVED** [review]
Approving with a couple remaining minor comments

File: projects/reservoir/fv3/era_process.py
Comment (paulaharder) [reply projects/reservoir/fv3/era_process.py:79]: That didn't work, we tried it a couple weeks ago

**Pull request merged**
